### ISSUE TITLE
Imports should be logged 

### ISSUE THREAD

**Issue by miloush** [Microsoft/msbuild#1437]
At least in diagnostic logging, `<Import/>` elements should be output (including condition evaluation if skipped).

**Comment by rainersigwald** [comment]
I've been thinking about this as part of #1331, because it will be _very_ noisy.

Note that you can get this information today by using the preprocessor, which evaluates conditions and shows all imports with their full path.

**Comment by jeffkl** [comment]
@miloush we're thinking about updating the output of the preprocessor which currently shows what's being imported but does not fully explain why an import is left out.  So the current output looks like this:

```xml
<!--<Import Project="SomeProject" Condition=" '(SomeProperty)' == 'true' And Exists('SomeProject') " />-->
```

And it would instead say:
```xml
<!--
<Import Project="SomeProject" Condition=" '(SomeProperty)' == 'true' And Exists('SomeProject') " />
 Not imported because the condition evaluated to " '' == 'true' And false "
-->
```

Would that suffice?  Putting it in the logger output would be very noisy and we think using the preprocessor is better for troubleshooting missing imports.

**Comment by miloush** [comment]
Yes, that looks like it would do for what I was after!

**Comment by jeffkl** [comment]
Closed by 5d44f341e1dd3a5149e13144ba00d36030f87ff4
